repository/src/main/resources/db/useful-queries/Useful select queries.sql

```diff
@@ -0,0 +1,95 @@
+-- ==========================================
+-- Details about tables within ccd-data-store
+-- ==========================================
+-- PK/FK relationships in tables case_data/case_event
+-- case_data pk column = id
+-- case_event pk column = id, fk column = case_data_id
+-- case_event_significant_items, pk column = id
+-- case_users, pk combined columns = case_data_id + user_id + case_role
+-- all_events, no pk or fk relationships defined
+
+-- =================================================================================
+-- Queries below return data from the various tables within ccd-data-store
+-- The select queries are ordered such that they can be used to delete ALL data by a
+-- given Jurisdiction. 
+-- I.E to perform a delete replace 'SELECT *' with 'DELETE'. Replace 'EMPLOYMENT' 
+-- with your Jurisdiction.
+
+    SELECT * FROM case_users_audit WHERE case_data_id IN (
+        SELECT id FROM case_data WHERE jurisdiction = 'EMPLOYMENT'
+    );
+
+    SELECT * FROM case_users WHERE case_data_id IN (
+        SELECT id FROM case_data WHERE jurisdiction = 'EMPLOYMENT'
+    );
+
+    SELECT * FROM case_event_significant_items WHERE case_event_id IN (
+        SELECT id FROM case_event WHERE case_data_id IN (
+            SELECT id FROM case_data WHERE jurisdiction = 'EMPLOYMENT'
+        )
+    );
+
+    SELECT *  FROM case_event WHERE case_data_id IN (
+        SELECT id FROM case_data WHERE jurisdiction = 'EMPLOYMENT'
+    );
+
+    SELECT id FROM case_event WHERE case_data_id IN (
+        SELECT id FROM case_data WHERE jurisdiction = jurisdiction_ref
+    );
+
+    SELECT * FROM all_events WHERE case_id IN (
+        SELECT id FROM case_data WHERE jurisdiction = 'EMPLOYMENT'
+    ) AND case_event_id in (
+        SELECT id  FROM case_event WHERE case_data_id IN (
+            SELECT id FROM case_data WHERE jurisdiction = 'EMPLOYMENT'
+        )
+    );
+
+    SELECT * FROM case_data WHERE jurisdiction = 'EMPLOYMENT';
+
+    SELECT COUNT(*) FROM case_data WHERE jurisdiction = 'EMPLOYMENT';
+-- =================================================================================
+
+-- List cases older than 3 Months, retaining highest version
+SELECT cd.id, cd.case_type_id, cd.version FROM case_data cd INNER JOIN
+        (SELECT reference, MAX("version") AS MaxVersion
+            FROM case_data
+            GROUP BY reference) grouped_cd
+    ON cd.reference = grouped_cd.reference
+    AND (cd.version != grouped_cd.MaxVersion AND 
+		 cd.created_date <= 'now'::timestamp - '3 MONTHS'::INterval)
```

Comment:
both case id and reference columns are unique, so this one didn't make full sense to me.